plugins/content_distribution/providers/tvinci/lib/api/KalturaTvinciDistributionJobProviderData.php

```diff
@@ -35,10 +35,11 @@ public function __construct(KalturaDistributionJobData $distributionJobData = nu
 
 		$feedHelper = new TvinciDistributionFeedHelper($distributionJobData->distributionProfile, $fieldValues);
 		$feedHelper->setEntryId( $entry->getId() );
+		$feedHelper->setReferenceId($entry->getReferenceID());
```

Comment:
Instead of all these setters, why won't you just setEntryDistribution($distributionJobData->entryDistribution), it contains everything you might need, including entry id.